examples/core/Graphormer/dataset.py

```diff
@@ -0,0 +1,107 @@
+"""
+This file contains the MolHIVDataset class, which handles data preprocessing
+(computing required graph features, converting graphs to tensors) of the
+ogbg-molhiv dataset.
+"""
+import torch as th
+import torch.nn.functional as F
+from dgl import shortest_dist
+from ogb.graphproppred import DglGraphPropPredDataset
+from torch.nn.utils.rnn import pad_sequence
+
+
+class MolHIVDataset(th.utils.data.Dataset):
+    def __init__(self):
+        dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
+        split_idx = dataset.get_idx_split()
+
+        # compute the shortest path distance during preprocessing
+        for g, label in dataset:
+            spd, path = shortest_dist(g, root=None, return_paths=True)
+            g.ndata["spd"] = spd
+            g.ndata["path"] = path
+
+        self.train, self.val, self.test = (
+            dataset[split_idx["train"]],
+            dataset[split_idx["valid"]],
+            dataset[split_idx["test"]],
+        )
+
+    def collate(self, samples):
+        # To be consistent with the input style of Graphormer, all graph
+        # features need to be padded into the same size (different number
+        # of nodes may make them inconsistent, and we align them with max
+        # number of nodes).
```

Comment:
Are you saying the features size should be the same, so we align features of all nodes with the node with max size of feature?

It is confusing with "we align them with max number of nodes".

Reply:
If so:

To match Graphormer's input style, all graph features should be padded to the same size. Keep in mind that different nodes may have varying feature sizes, so they will be aligned with the node having the largest feature size.

Reply:
Sorry for this confusing comment. Actually, I want to say that "we align them with the **graph** having maximum number of nodes". For example, node feature of a graph has shape `[n, d]`, where `n` is the number of nodes in that graph and `d` is the dimension of each node feature. `n` may vary for different graphs while the input style required by Graphormer is `[b, n, d]` with batch size `b`. Thus node features should be padded to `[N, d]`, where `N` is the maximum number of nodes among graphs from this batch. In this way, node features from different graphs have the same shape and can be stacked as the input with shape `[b, N, d]`.

How about:
To match Graphormer's input style, all graph features should be padded to the same size. Keep in mind that different graphs may have varying feature sizes since they have different number of nodes, so they will be aligned with the graph having the maximum number of nodes.